doc/source/user_guide/plots_arguments_guide.md

```diff
@@ -83,6 +83,21 @@ Use `var_names` to indicate which variables to exclude with the `~` prefix:
 az.plot_posterior(centered_eight, var_names=['~mu', '~theta']);
 ```
 
+Variables do not need to be strings to be used
```

Comment:
What are the conditions needed for that to work? Am I right to assume they are only two:
1. Having `__repr__` defined
2. Being a hashable (aka [valid xarray variable name](https://xarray.pydata.org/en/stable/user-guide/terminology.html#term-name))

which brings to the 2nd question. Could this example use int or tuple variable names instead of defining a custom class?

Reply:
and maybe 3rd question, if the `~` were to only work on strings, could the condition be reduced to being a valid xarray name only? it looks like the _repr helper is only used right before using `startswith`, maybe is could be updated to be `check_tilde_start` or something like this to always return False if not str (I think, it is late and my logic and double negations already went to sleep, I only know it would need to be constant), check `.startswith` if str.